static/src/javascripts/projects/common/modules/atoms/youtube.js

```diff
@@ -252,55 +263,84 @@ const STATES = {
     PAUSED: onPlayerPaused,
 };
 
-const getIFrameBehaviour = (atomId: string): IFrameBehaviour => {
-    const isAutoplayBlockingPlatform = () => isIOS() || isAndroid();
+const getIFrameBehaviourConfig = (
+    iframe: HTMLIFrameElement
+): IFrameBehaviourConfig => {
+    const isAutoplayBlockingPlatform = isIOS() || isAndroid();
 
-    const isInternalReferrer = (): boolean => {
+    const isInternalReferrer = ((): boolean => {
         if (config.get('page.isDev')) {
             return document.referrer.indexOf(window.location.origin) === 0;
         }
 
         return document.referrer.indexOf(config.get('page.host')) === 0;
-    };
+    })();
 
-    const isMainVideo = (): boolean =>
-        (players[atomId].iframe &&
-            !!players[atomId].iframe.closest(
-                'figure[data-component="main video"]'
-            )) ||
+    const isMainVideo =
+        (iframe && !!iframe.closest('figure[data-component="main video"]')) ||
         false;
 
-    const flashingElementsAllowed = (): boolean =>
-        accessibilityIsOn('flashing-elements');
+    const flashingElementsAllowed = accessibilityIsOn('flashing-elements');
 
-    const isVideoArticle = (): boolean =>
+    const isVideoArticle =
         config.get('page.contentType', '').toLowerCase() === 'video';
 
-    const isFront = () => config.get('page.isFront');
+    const isFront = config.get('page.isFront');
```

Comment:
This predates your changes, however IIRC `config.get` takes an optional second parameter that is used in the event that value of the key is `undefined`:
https://github.com/guardian/frontend/blob/master/static/src/javascripts/lib/config.js#L12

So one minor improvement would be to use `config.get('page.isFront', false)` or `!!config.get('page.isFront')`